### ISSUE TITLE
Suggestion/Question about "projects" and groups

### ISSUE THREAD

**Issue by It4lik** [vmware/harbor#1305]
Hi, 
First of all, thanks for this great service. 

I wanted to know if you plan to add a group-like functionality. I know this isn't really needed since "projects" are more or less a group functionality. But in certain circumstances, it should be better to have something like groups : 
  - if I want to share an image with about fifty collaborators, I need to add them one by one in the parent project of that specific image. That can be a bit painful. With groups-like functionality, we would be able to share an image with a whole group of collaborators. 
  - I think that the first authentication of a user (on the WebUI or _via_ `docker login` ) should automatically creates a named project using the user's name.
  - I tested out a bit Atomic Registry (from RedHat) and I found their "LDAP group syncing" functionality really cool (maybe a bit more advanced). 

These are just thoughts and suggestions :)
It4.

**Comment by reasonerjt** [comment]
@It4lik 
Let me confirm, you suggest that we introduce more flexibility in the permission model, that a "namespace" can be granted with read/write permission to a group of user?  So will the users in one group have the same read/write permission under the same namespace?

**Comment by It4lik** [comment]
Yep, instead of adding users one by one to a project (and the fact that we must know the EXACT name is a pain too, since there is no searchbar for users or autocompletion when typing first letters)

**Comment by palonsoro** [comment]
In large companies, it is a very common pattern to define groups of people and give permissions to those groups, so whenever new staff joins or leaves the company or that department, you only have to add/delete people from a single place (one group) and not N projects (all the projects that group must be granted access to).

**Comment by hbastosc** [comment]
Hi, is there any plans to implement this feature in the near future?

**Comment by GurvanLd** [comment]
+1 it's a headache I have to add manually each my 100 users the permission to access projects ...

**Comment by reasonerjt** [comment]
This is being worked on https://github.com/vmware/harbor/issues/3506
@hbastosc @GurvanLd
